### PR TITLE
move style/ and lib/ to src/

### PR DESCRIPTION
**Pull request by ngokevin**
Clean up root folder. This css counts as source code.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -7,8 +7,8 @@ if (!('import' in document.createElement('link'))) {
   require('../lib/vendor/HTMLImports');
 }
 
-require('../style/aframe-core.css');
-require('../style/rStats.css');
+require('./style/aframe-core.css');
+require('./style/rStats.css');
```
Comment (cvan) [inline src/index.js]: you need to fix this path for rStats:

https://github.com/aframevr/aframe/blob/dev/src/core/a-scene.js#L621

File: src/index.js [hunk 2]
```diff
@@ -7,8 +7,8 @@ if (!('import' in document.createElement('link'))) {
   require('../lib/vendor/HTMLImports');
 }
 
-require('../style/aframe-core.css');
-require('../style/rStats.css');
+require('./style/aframe-core.css');
```
Comment (cvan) [inline src/index.js]: you can remove [`style` from the budo script watch list](https://github.com/aframevr/aframe/blob/dev/scripts/budo.js#L19)

**Comment by cvan** [comment]
related: shouldn't [`lib/`](https://github.com/aframevr/aframe/tree/dev/lib) also be in `src/`?

**Comment by ngokevin** [comment]
Yeah i'll move lib to src. Okay will fix.

**Comment by cvan** [comment]
might want to change name of PR

**Comment by cvan** [comment]
don't forget to remove the [old directories](https://github.com/aframevr/aframe/blob/dev/package.json#L15) from the `preghpages` npm script

**Comment by cvan** [comment]
@ngokevin: [linting failed](https://travis-ci.org/aframevr/aframe/jobs/102203877)

you'll need to adjust the `semistandard` key in the `package.json`

**Comment by cvan** [comment]
do you have the husky pre-commit hook enabled?

**Comment by ngokevin** [comment]
Yeah, i'm working on the linting now. i have the precommit hook in my .git/hooks, but it doesn't do anything.

**Comment by cvan** [comment]
> Yeah, i'm working on the linting now. i have the precommit hook in my .git/hooks, but it doesn't do anything.

how come? if you do a fresh install, does it? has it ever?

**Comment by ngokevin** [comment]
ok i might've accidentally disabled it during the merge. i was trying to merge and was getting linting errors unrelated to me.

File: package.json [hunk 3]
```diff
@@ -10,7 +10,7 @@
     "dist": "mkdirp dist/ && npm run browserify -s -- --debug | exorcist dist/aframe.js.map > dist/aframe.js && uglifyjs dist/aframe.js -c warnings=false -m -o dist/aframe.min.js",
     "gh-pages": "npm run ghpages",
     "ghpages": "node ./scripts/gh-pages",
-    "lint": "semistandard -v $(git ls-files '*.js') | standard-reporter --stylish",
+    "lint": "semistandard -v $(git ls-files '*.js' | grep -v src/lib/) | standard-reporter --stylish",
```
Comment (dmarcos) [inline package.json]: this won't work on windows

Comment (cvan) [inline package.json]: @ngokevin just change [this line](https://github.com/ngokevin/aframe/blob/stylesrc/package.json#L86) to be `src/lib/…`

Comment (ngokevin) [inline package.json]: oh thanks

**Comment by dmarcos** [comment]
What's the purpose of moving ```lib``` into ```src```. When I'm editing code I don't want to lib to get in the way. I don't want to encourage people to sumbint PRs against ``lib```. ```src``` should contain just code that is developed and maintained by us

**Comment by ngokevin** [comment]
I don't think there's a difference between keeping it in the root vs the src folder. People have already tried to submit PRs against `aframe/lib`. It just cleans up the root folder and requires because it was kind of weird to have src modules locally reach outside of the src directory.

**Comment by dmarcos** [comment]
There is. There are tons of projects that follow a separate src/lib directory and it makes sense to keep a clear distinction between them.

**Comment by ngokevin** [comment]
Naming the folder `lib` already creates the distinction.

**Comment by ngokevin** [comment]
A lot of projects also use `lib` as a dist folder or built code:

https://github.com/CreateJS/TweenJS/tree/master/lib
https://github.com/ngokevin/react-router-reverse/tree/master/lib

**Comment by cvan** [comment]
rebase pls

File: scripts/budo.js [hunk 4]
```diff
@@ -16,7 +16,7 @@ var consts = {
   ENTRY: './src/index.js',
   DIST: 'dist/aframe-core.js',
   BUILD: 'build/aframe-core.js',
-  WATCH: '{examples,lib,src,style}/**/*',
+  WATCH: '{examples,src}/**/*',
```
Comment (cvan) [inline scripts/budo.js]: can just do `{examples,src/lib}/**/**`

File: package.json [hunk 5]
```diff
@@ -83,7 +83,7 @@
       "build/**",
       "dist/**",
       "examples/_js/**",
-      "lib/vendor/**"
+      "src/lib/**"
```
Comment (cvan) [inline package.json]: I'd keep as `src/lib/vendor/**`

**Comment by cvan** [comment]
r+ works well, thanks

**Pull request merged**